src/ConnectedMode/Persistence/SolutionBindingRepository.cs

```diff
@@ -100,7 +100,11 @@ public bool Write(string configFilePath, BoundServerProject binding)
         return true;
     }
 
-    public bool DeleteBinding(string configFilePath) => throw new NotImplementedException();
+    public bool DeleteBinding(string localBindingKey)
+    {
+        var bindingPath = unintrusiveBindingPathProvider.GetBindingPath(localBindingKey);
```

Comment:
Not required, but a question. [Suppressions](https://github.com/SonarSource/sonarlint-visualstudio/blob/f16c3a1f8dfab8ade4df0af546a0dc2a9d976a4d/src/Roslyn.Suppressions/Roslyn.Suppressions/SuppressionExecutionContext.cs#L34) check for binding to be present and do nothing if there's no connected mode. We delete the binding here, and the suppressions will stop working (well, there might be issues with roslyn caching stuff, but that's not caused by any of this). But we still keep the suppressions file in `AppData\Local\Temp\SLVS\Roslyn`. Could you check if there are no unintended consequences on rebind? The file should get replaced instantly, but still it would be good to double check.

Reply:
I tested with the Suppressions_CSharp.cs file from our sample projects the following cases:

Case 1:
1. Bind to Samples_Bound_2019 (suppressions are there) 
2. Unbind (suppressions are removed)
3. Bind again to Samples_Bound_2019 (suppressions are there) 

=> This case seems to be working correctly

Case 2:
1. Bind to Samples_Bound_2019 (suppressions are there) 
2. Unbind (suppressions are removed)
3. Bind to another project that is NOT Samples_Bound_2019 and that does not have the suppressions in this file

=> The old suppressions seem to be applied. But they disappear as soon as analysis of the file is triggered (reopening the file)

This seems to be caused by the fact that the old file is there for a brief period of time before being replaced. Should we also consider deleting the file from %tmp%?

Reply:
yeah I think it would be safer to delete it as well. you can move it to another PR